src/OrchardCore/OrchardCore.Email.Core/Services/SmtpEmailProvider.cs

```diff
@@ -0,0 +1,333 @@
+/// </summary>
+public class SmtpEmailProvider : IEmailProvider
+{
+    public const string TechnicalName = "SMTP";
+
+    private const string EmailExtension = ".eml";
+
+    private static readonly char[] _emailsSeparator = [',', ';'];
+
+    private readonly SmtpOptions _providerOptions;
+    private readonly ILogger _logger;
+    protected readonly IStringLocalizer S;
+
+    /// <summary>
+    /// Initializes a new instance of a <see cref="SmtpEmailProvider"/>.
+    /// </summary>
+    /// <param name="options">The <see cref="IOptions{SmtpSettings}"/>.</param>
+    /// <param name="logger">The <see cref="ILogger{SmtpService}"/>.</param>
+    /// <param name="stringLocalizer">The <see cref="IStringLocalizer{SmtpService}"/>.</param>
+    public SmtpEmailProvider(
+        IOptions<SmtpOptions> options,
+        ILogger<SmtpEmailProvider> logger,
+        IStringLocalizer<SmtpEmailProvider> stringLocalizer)
+    {
+        _providerOptions = options.Value;
+        _logger = logger;
+        S = stringLocalizer;
+    }
+
+    /// <summary>
+    /// The name of the provider.
+    /// </summary>
+    public LocalizedString Name => S["Simple Mail Transfer Protocol (SMTP)"];
+
+    /// <summary>
+    /// Sends the specified message to an SMTP server for delivery.
+    /// </summary>
+    /// <param name="message">The message to be sent.</param>
+    /// <returns>A <see cref="EmailResult"/> that holds information about the sent message, for instance if it has sent successfully or if it has failed.</returns>
+    /// <remarks>This method allows to send an email without setting <see cref="MailMessage.To"/> if <see cref="MailMessage.Cc"/> or <see cref="MailMessage.Bcc"/> is provided.</remarks>
+    public async Task<EmailResult> SendAsync(MailMessage message)
+    {
+        if (!_providerOptions.IsEnabled)
+        {
+            return EmailResult.FailedResult(S["The SMTP Email Provider is disabled."]);
+        }
+
+        EmailResult result;
+        var response = default(string);
+        try
+        {
+            // Set the MailMessage.From, to avoid the confusion between _options.DefaultSender (Author) and submitter (Sender)
+            var senderAddress = string.IsNullOrWhiteSpace(message.From)
+                ? _providerOptions.DefaultSender
+                : message.From;
+
+            if (!string.IsNullOrWhiteSpace(senderAddress))
+            {
+                message.From = senderAddress;
+            }
+
+            var errors = new List<LocalizedString>();
+
+            var mimeMessage = FromMailMessage(message, errors);
+
+            if (errors.Count > 0)
+            {
+                return EmailResult.FailedResult(errors.ToArray());
+            }
+
+            if (mimeMessage.To.Count == 0 && mimeMessage.Cc.Count == 0 && mimeMessage.Bcc.Count == 0)
+            {
+                return EmailResult.FailedResult(S["The mail message should have at least one of these headers: To, Cc or Bcc."]);
```

Comment:
These validations are also necessary for Azure, see https://github.com/OrchardCMS/OrchardCore/blob/83aae0a63efdde0d692d237c31b16858ac1535c2/src/OrchardCore/OrchardCore.Email.Core/Services/EmailMessageValidator.cs

Reply:
This kind of validation should not be part of the provider. It should be part of the form that collects that info.

Reply:
I still don't think we should do email validation in the provider level. We should not have to ask every provider to do it's validation. 

If we don't want to focus validation in the forms, then maybe we should add a new event handler for validating so this way we can use event handle to do this kind of validation in the service and before calling the providers.

Reply:
I'm OK with having this validation in EmailService, just not only in SmtpProvider.